### ISSUE TITLE
oneflow.amax perform differently between cpu and cuda when dim=1

### ISSUE THREAD

**Issue by Redmept1on** [Oneflow-Inc/oneflow#10473]
## Summary

oneflow.softmax perform differently between cpu and cuda when dim=1

## Code to reproduce bug

```python
import oneflow as flow
import numpy as np

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cuda()
y1 = flow.amax(x1,dim=1)
print(y1.device,y1)

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cpu()
y2 = flow.amax(x1,dim=1)
print(y2.device,y2)
```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/396556da-c1dd-47a6-85fc-3078d6ae4d84)

pytorch
```python
import torch
import numpy as np

input_tensor = torch.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
# other_tensor = torch.tensor(np.array([0, 1, 1], dtype=np.float32))
output_tensors = torch.amax(input_tensor,dim=1)

print(output_tensors)
```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/a43bed86-8be0-4eb4-baae-f422cd8b4ad5)

## System Information

- What is your OneFlow installation :dockerhub
- OS:Linux
- OneFlow version (run python3 -m oneflow --doctor):0.9.1.dev20231024+cu117
- Python version: 3.8

**Comment by Redmept1on** [comment]
oneflow.amin has the same issue
```python
import oneflow as flow
import numpy as np

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cuda()
y1 = flow.amin(x1,dim=1)
print(y1.device,y1)

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cpu()
y2 = flow.amin(x1,dim=1)
print(y2.device,y2)

```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/4cebfbc1-9429-4a63-8b50-b3b40267a0dd)
